include/boost/compute/exception/program_build_failure.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef BOOST_COMPUTE_EXCEPTION_PROGRAM_BUILD_FAILURE_HPP
+#define BOOST_COMPUTE_EXCEPTION_PROGRAM_BUILD_FAILURE_HPP
+
+#include <string>
+
+#include <boost/compute/exception/opencl_error.hpp>
+
+namespace boost {
+namespace compute {
+
+/// \class program_build_failure
+/// \brief A failure when building OpenCL program
+///
+/// Instances of this class are thrown when OpenCL program build fails.
+/// Extends opencl_error by saving a program build log so it can be used for
+/// testing/debugging/logging purposes.
+///
+/// \see opencl_error
+class program_build_failure : public opencl_error
+{
+public:
+    /// Creates a new program_build_failure exception object for \p error
+    /// and \p build_log.
+    explicit program_build_failure(cl_int error, const std::string& build_log)
+        throw()
+        : opencl_error(error),
+          m_build_log(build_log)
+    {
+    }
+
```

Comment:
You need to explicitly add empty destructor with `throw()` specifier to make it `noexcept`. Destructors are implicitly `noexcept` since C++11.